### DISCUSSION TITLE
Issue with Displaying and Saving Related Fields in CRUD Edit Screen

### DISCUSSION THREAD

**Discussion by infiniteroles** [Laravel-Backpack/community-forum#1142] [Q&A (Help)] [answered]
Hello,

I’m still having an issue with displaying and saving related fields in my CRUD interface. I think my previous explanation may not have been clear, so I’ll try again.

I have two models:
1. User model:
`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use App\Observers\UserObserver;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Fortify\TwoFactorAuthenticatable;
use Laravel\Jetstream\HasProfilePhoto;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Support\Str;
use Illuminate\Support\Carbon;
use Illuminate\Support\Facades\DB;
use Vinkla\Hashids\Facades\Hashids;
use App\Notifications\RecoverPasswordNotification;
use Illuminate\Support\Facades\Auth; 

class User extends Authenticatable implements MustVerifyEmail
{
    use CrudTrait;
    //use HasApiTokens;
    use HasFactory;
    use HasProfilePhoto;
    use Notifiable;
    use TwoFactorAuthenticatable;
    use UserReferral;
    use UserPromotionalCode;
    // Borrado usuarios
    use SoftDeletes;

    // Admin Role ID number
    public $adminID = 100;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'id',
        'name',
        'email',
        'onboarding_id',
        'contact_permission_level',
        'risk',
        'risk_date',
    ];

    public function userType()
    {
        return $this->belongsTo(UserType::class);
    }`

2.AccountVerifciation model:

`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

class AccountVerification extends Model
{
    use CrudTrait;
    use HasFactory;

    /**
     * TABLE NAME
     *
     */
    public $table = "account_verifications";

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'user_id',
        'date',
        'status',
        'type',
        'name',
        'surname',
        'surname2',
        'dni',
        'dni_expiration_date',
        'date_birth',
        'phone',
        'address',
        'city',
        'postal_code',
        'country',
    ];

    public function user()
    {
        return $this->belongsTo(User::class)->withTrashed();
    }

    public function getUserEmailAttribute()
    {
        return $this->user->email;
    }

    public function getUserNameAttribute()
    {
        return $this->user->name;
    }
`

I want to display the “risk” field from the “User” model in the AccountVerification CRUD edit view. Additionally, I need any changes made to this “risk” field within the AccountVerification edit view to be correctly saved back to the database.

Currently, I have implemented it as follows in my CRUD setup, but the changes to the “risk” field are not being saved.

`<?php

namespace App\Http\Controllers\Admin;

use App\Http\Requests\AccountVerificationRequest;
use Backpack\CRUD\app\Http\Controllers\CrudController;
use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
use Backpack\CRUD\app\Library\Widget;

/**
 * Class AccountVerificationCrudController
 * @package App\Http\Controllers\Admin
 * @property-read \Backpack\CRUD\app\Library\CrudPanel\CrudPanel $crud
 */
class AccountVerificationCrudController extends CrudController
{
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;

    /**
     * Configure the CrudPanel object. Apply settings to all operations.
     * 
     * @return void
     */
    public function setup()
    {
        CRUD::setModel(\App\Models\AccountVerification::class);
        CRUD::setRoute(config('backpack.base.route_prefix') . '/account-verification');
        CRUD::setEntityNameStrings('KYC', 'KYC');
        CRUD::denyAccess('show');
    }

    /**
     * Define what happens when the List operation is loaded.
     * 
     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
     * @return void
     */
     
         /**
     * Define what happens when the Create operation is loaded.
     * 
     * @see https://backpackforlaravel.com/docs/crud-operation-create
     * @return void
     */
    protected function setupCreateOperation()
    {
        CRUD::addField([
            'name' => 'user_id',
            'type' => 'text',
            'label' => 'User',
            'entity' => 'user',
            'attribute' => 'name',
            'model' => "App\Models\User",
            'wrapper' => ['class' => 'col-md-1'],
            'attributes' => ['readonly' => 'readonly'],
        ]);
                CRUD::addField([
            'name' => 'risk',
            'label' => 'Risk',
            'type' => 'select2_from_array',
            'entity' => 'user',
            'attribute' => 'risk',
            'allows_null' => false,
            'model' => "App\Models\User",
            'options' => [
                'BAJO' => 'BAJO',
                'MEDIO' => 'MEDIO',
                'ALTO' => 'ALTO',
            ],
            'wrapper' => ['class' => 'col-md-1'],
            ]); 
            .-......
`

Thank you very much!!!!

Oscar

**Comment by jcastroa87** [comment, marked as answer]
Hello @infiniteroles 

First you need to add "risk" to the fillable on "AccountVerification" model.

Then to handle it i recommend use Accessor and Mutator in "AccountVerification" model, like this:
```
public function getRiskAttribute()
    {
        return $this->user ? $this->user->risk : null;
    }

    public function setRiskAttribute($value)
    {
        if ($this->user) {
            $this->user->risk = $value;
            $this->user->save();
        }
    }
```

Finally, with this set, you can simplify your controller field:
```
CRUD::addField([
            'name' => 'risk',
            'label' => 'Risk',
            'type' => 'select2_from_array',
            'allows_null' => false,
            'options' => [
                'BAJO' => 'BAJO',
                'MEDIO' => 'MEDIO',
                'ALTO' => 'ALTO',
            ],
            'wrapper' => ['class' => 'col-md-1'],
        ]);
```

This should work, let me know if you need any further help.

Cheers.

**Reply by infiniteroles** [reply]
Thanks!! that worker perfect!!

**Comment by jcastroa87** [comment]
Im glad this work for you.

I will close the issue now, but please feel free to re-open or create a new one.

Cheers.
